bokehjs/src/lib/models/annotations/title.ts

```diff
@@ -131,8 +158,8 @@ export class Title extends TextAnnotation {
       ["background_", mixins.Fill],
     ])
 
-    this.define<Title.Props>(({Number, String}) => ({
-      text:             [ String, "" ],
+    this.define<Title.Props>(({Number, String, Or, Ref, Nullable}) => ({
+      text:             [ Nullable(Or(String, Ref(BaseText))), null],
```

Comment:
Can a title text be null?

Reply:
Don't know. I'll need to check.

Reply:
It's `Plot.title` that can be `null`. This will be always initialized.

Reply:
Thanks, that makes it simpler.

Reply:
I think the choice here is between
```typescript
text: [ Nullable(Or(String, Ref(BaseText))), null ], 
```
and
```typescript
text: [ Or(String, Ref(BaseText)), "" ], 
```
The former is easier to check (`text != null`) than the latter as we can use `String.length` but there is no `length` for `BaseText`.

Reply:
if you want to check if there is text you can simply use `if (text)` because an empty string if falsy.
![image](https://user-images.githubusercontent.com/26139868/134908339-fd657571-f0cb-43d8-a98d-d914397dbd3f.png)